### PR TITLE
FIX #71: old symlink should be removed on Windows

### LINKED ISSUE #71: [BUG] Windows symlink bug
platform Win10
zvm v0.6.1 windows/amd64

after zvm ls
```
2024/04/09 21:59:10 WARN exec: "zig": executable file not found in %PATH%
0.10.1
0.11.0
master
```
The path and env are configured properly,

After zvm use 0.11.0  (or any other version) , there is following error :
```
2024/04/09 21:53:40 ERRO remove C:\Users\francek\.zvm\bin: The system cannot find the file specified.
```
There is actualy no .zvm\bin subdir

### PR DESCRIPTION
**Pull request by guidoschmidt**
Seems to be a longer journey 😅 ...

just tested version 0.6.2 and unfortunately the old symlink to `.zvm/bin` is not removed, ~~the change to `!errors.Is(err, os.ErrNotExist)` seems to fix it.~~ found `os.Lstat` which checks symbolic links.

Maybe there's a more appropriate way to do this? Looks like there's a *newer* [fs package](https://pkg.go.dev/io/fs#Stat) that can be used for these kind of filesystem checks.

### CODE REVIEW

**Comment by tristanisham** [comment]
Thanks for making another PR. These system specific bugs are always a pain but we'll fix this one. I'll be on my Windows developer environment tonight so I can work on this bug directly.

File: cli/use.go [hunk 1]
```diff
@@ -40,9 +39,11 @@ func (z *ZVM) setBin(ver string) error {
...
 	bin_dir := filepath.Join(z.baseDir, "bin")
 
-	// Remove "bin" dir only if it already exists
-	
-	if _, err := os.Stat(bin_dir); errors.Is(err, fs.ErrExist) {
+	// Came across https://pkg.go.dev/os#Lstat
+	// which is specifically to check symbolic links.
+	// Seemed like the more appropriate solution here
+	stat, err := os.Lstat(bin_dir);
+	if errors.Is(err, os.ErrExist) || stat != nil {
```
[the remark this answers is not in the document]
Comment (guidoschmidt) [reply cli/use.go:46]: I think you can't mix up the `os` with the `fs` stuff, as they are different packages? But it's only my educated guess. Honestly it feels like there's way too many ways to do this in golang 😓 the best way maybe is to use `os.Lstat` and just see if the link is nil? 🤔

**Comment by guidoschmidt** [comment]
Oh nice, hope it doesn't give you too much of a headache

**Comment by tristanisham** [comment]
New bug unlocked. Totally breaks the progress bar when running in command prompt. 
image

**Comment by tristanisham** [comment]
@guidoschmidt could you please test the changes I've pushed to the PR. For I modified the Symlink function again because for users who are already admins or have enabled symlinks for regular users it was uneccessary and degraded the experience. I also tweaked the Lstat to use the err == nil method since just parsing the errors wasn't working for us.

**Comment by tristanisham** [comment]
Added a test for our symlink code. It passes so that bit in setBin should be good.
```go
func TestSymlinkExists(t *testing.T) {
	if err := os.Symlink("use_test.go", "symlink.test"); err != nil {
		t.Error(err)
	}

	stat, err := os.Lstat("symlink.test");
	if err != nil {
		t.Errorf("%q: %s", err, stat.Name())
	}

	defer os.Remove("symlink.test")
}
```

**Pull request merged**